addons/mail/static/src/core/common/store_service.js

```diff
@@ -180,12 +180,15 @@ export class Store extends BaseStore {
         return tab === "chat" ? ["chat", "group"] : [tab];
     }
 
-    handleClickOnLink(ev, thread) {
+    async handleClickOnLink(ev, thread) {
         const model = ev.target.dataset.oeModel;
         const id = Number(ev.target.dataset.oeId);
         if (ev.target.closest(".o_channel_redirect") && model && id) {
             ev.preventDefault();
             const thread = this.Thread.insert({ model, id });
+            if (!thread.is_pinned) {
+                await this.env.services["mail.thread"].fetchChannel(id);
+            }
```

Comment:
Don't turn in async, instead do a `fetchChannel.then(open)` while returning `true` immediately.

The function is designed to return synchronously whether the click is handled or not.
This is important to keep the mouse event `ev` markings and state handling as immediate as possible.